src/package.rs

```diff
@@ -1,35 +1,50 @@
 //! RPM package type: represents `.rpm` files or entries in the RPM database
 use std::fmt;
 
+use crate::internal::{header::Header, tag::Tag};
+
 /// RPM packages
 #[derive(Clone, Debug, Eq, PartialEq, Hash)]
 pub struct Package {
     /// Name of the package
-    pub(crate) name: String,
+    name: String,
 
     /// Epoch of the package
-    pub(crate) epoch: Option<String>,
+    epoch: Option<String>,
 
     /// Version of the package
-    pub(crate) version: String,
+    version: String,
 
     /// Release of the package
-    pub(crate) release: String,
+    release: String,
 
     /// Arch of the package
-    pub(crate) arch: String,
+    arch: String,
 
     /// License of the package
-    pub(crate) license: String,
+    license: String,
 
     /// Succinct description of the package
-    pub(crate) summary: String,
+    summary: String,
 
     /// Longer description of the package
-    pub(crate) description: String,
+    description: String,
 }
 
 impl Package {
+    pub(crate) fn from_header(h: &Header) -> Self {
```

Comment:
Since we're taking a look at this one again, I think it might make sense to make this a `Result<Self, ...>` and propagating the errors below, rather than using `unwrap()`